### ISSUE TITLE
Google Translate unload event listener

### ISSUE THREAD

**Issue by jduss4** [livingstoneonline/onemorevoice#2]
I have not worked much with event listeners and so perhaps I am missing something obvious, but I thought the first step to solving this problem that you say was introduced when you added Google Translate would be to go figure out why GT was registering an unload listener.

I had trouble finding a current google API for adding translation services to a website in the way that you had, so then I kind of petered out in terms of solving the problem. I guess I'm basically punting on this one because I don't know enough about any of these things to give you good counsel, but it seems like you could probably try messing with `window.removeEventListener("unload", someFunction)` and see if that changes the lighthouse output.

**Comment by awisnicki** [comment]
Here's what I've been able to figure out with this issue. It's not much, but perhaps will be of some use. 

1. Here's the issue as Lighthouse flags it:

<img width="913" alt="Screen Shot 2021-01-27 at 13 09 11" src="https://user-images.githubusercontent.com/12518623/106041286-3a5b5f80-60a1-11eb-9120-6af6469d6905.png">

2. The specific Google JS file being referenced is this: [https://translate.googleapis.com/element/TE_20201130_00/e/js/element/element_main.js](https://translate.googleapis.com/element/TE_20201130_00/e/js/element/element_main.js). Also attached.
[element_main.txt](https://github.com/livingstoneonline/onemorevoice/files/5882491/element_main.txt)
 (Note: GitHub wouldn't accept a JS file, so I changed suffix .txt so be sure to change it back to .js).

3. If you open that file in a text editor and go to line 114, you'll find this code; the listener in question, I think, appears right at the beginning of it:

              function Lj() {
                  var a = Nj,
                      b = nj
                          ? function (c) {
                              return a.call(b.src, b.listener, c)
                          }
                          : function (c) {
                              c = a.call(b.src, b.listener, c);
                              if (!c) 
                                  return c
                          };
                  return b
              }
              function Hj(a, b, c, d, e) {
                  if (Array.isArray(b)) {
                      for (var f = 0; f < b.length; f++) 
                          Hj(a, b[f], c, d, e);
                      return null
                  }
                  c = Ij(c);
                  return wj(a)
                      ? a.Ab(b, c, tc(d)
                          ? !!d.capture
                          : !!d, e)
                      : Jj(a, b, c, !0, d, e)
              }
              function Oj(a, b, c, d, e) {
                  if (Array.isArray(b)) 
                      for (var f = 0; f < b.length; f++) 
                          Oj(a, b[f], c, d, e);
              else 
                      d = tc(d)
                          ? !!d.capture
                          : !!d,
                      c = Ij(c),
                      wj(a)
                          ? a.ga(b, c, d, e)
                          : a && (a = Kj(a)) && (b = a.Bc(b, c, d, e)) && Pj(b)
              }

That said, in the Google JS file, I see at least two instances of something related to "unload" being called out, so perhaps one of those is actually where this issue is? In any case, that definitely takes me to the limits of what I understand about this issue.

4. On any given page of my site the sequence of loading, roughly speaking, is this:
a. The page loads.
b. The Google translate stuff loads.
c. Finally, my main JS file loads ([minified version](https://github.com/livingstoneonline/onemorevoice/blob/master/js/scripts.js) | [unminified version](https://github.com/livingstoneonline/onemorevoice/blob/master/js/scripts-dev.js)). Each page has a script at the bottom, which you reference in this issue #8, to ensure that the minified version of that file loads absolute last in the sequence.

5. So, in short, it's my belief that it's just a matter of adding some code to my main JS file that turns off or otherwise overrides that event listener in the Google code since the JS file will come last in the loading sequence after the Google code. The issue, however, is that even listeners are definitely beyond my skill set, so that's about as far as I can take this myself.

**Comment by awisnicki** [comment]
I've made some progress on this issue. First, using Chrome dev tools, I was able to identify the problematic unload listener definitively:

```
    function Lj() {
        var a = Nj,
            b = nj
                ? function (c) {
                    return a.call(b.src, b.listener, c)
                }
                : function (c) {
                    c = a.call(b.src, b.listener, c);
                    if (!c) 
                        return c
                };
        return b
    }
```
And it appears in this file, which is loaded automatically by Google translate: https://translate.googleapis.com/element/TE_20201130_00/e/js/element/element_main.js

Moreover, in one of its test, Google Lighthouse itself tells me that this file is not needed!

<img width="557" alt="Screen Shot 2021-02-22 at 08 40 44" src="https://user-images.githubusercontent.com/12518623/108760883-d72fe200-7513-11eb-8778-3eecc0e0d696.png">

As a result, it's clear that not just the unload listener but the whole file can be removed from the DOM. So I came up with a script that does exactly that:

```
$('head').find('script').filter(function(){
    return $(this).attr('src') === 'https://translate.googleapis.com/element/TE_20201130_00/e/js/element/element_main.js'
}).remove();
```

In other words, Google loads the file, then this script removes it directly afterwards. However, Lighthouse still doesn't like this and is still flagging the unload listener! 

So, in short, it's not that I have to remove this from the DOM; rather I have to block it from ever appearing in the DOM in the first place. Unfortunately, I've not been able to figure out how to do that. 

Is there a way to preemptively block (using JavaScript or JQuery) any scripts that contain the src value "https://translate.googleapis.com/element/TE_20201130_00/e/js/element/element_main.js" from ever loading on the page/appearing in the DOM in the first place?

**Comment by jduss4** [comment]
Hi Adrian, sorry, I'm not ignoring you, just wrapping some things up this week!  I'll try to take a look tomorrow, but my instinct says that if you host all the JS you could just remove it....not an awesome practice, but it would work in this case.  I'll circle back around to this soon!

**Comment by awisnicki** [comment]
@jduss4 OK, sounds great. Yes, it's no problem to host this particular JS file myself, but as noted above in this case it turns out not to be needed at all, so essentially – as noted two comments above – I just need something that says if this the site tries to load this URL ("https://translate.googleapis.com/element/TE_20201130_00/e/js/element/element_main.js"), just block it.
